TypeCobol.Analysis/Graph/ControlFlowGraph.Dominator.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using TypeCobol.Analysis.Util;
+
+namespace TypeCobol.Analysis.Graph
+{
+    /// <summary>
+    /// Dominator algorithm implementation
+    /// </summary>
+    /// <typeparam name="N"></typeparam>
+    /// <typeparam name="D"></typeparam>
+    public partial class ControlFlowGraph<N, D>
+    {
+        /// <summary>
+        /// Compute a Dominator set on this CFG Graph
+        /// </summary>
+        /// <returns>A Tuple of a List of reachable nodes and an array of BitSet indexed by BasicBlock Indices</returns>
+        public Tuple<List<int>,BitSet[]> ComputeDominators()
+        {
+            BitSet[] dominators = new BitSet[this.AllBlocks.Count];
+            //Compute reachable blocks
+            List<int> blocks = new List<int>();
+            this.DFS((block, incomingEdge, predecessorBlock, cfg) =>
+            {
+                blocks.Add(block.Index);
+                dominators[block.Index] = new BitSet(dominators.Length);
+                if (block.Index != this.RootBlock.Index)
+                    dominators[block.Index].Full();
+                return true;
+            });
+            //Setup prececessors
+            this.SetupPredecessorEdgesFromRoot();
+
+            //Reflexivity on root block
+            dominators[this.RootBlock.Index].Set(this.RootBlock.Index);
+
+            bool change = false;
+            BitSet workSet = new BitSet(dominators.Length); //workingSet
+            do
+            {
+                change = false;
+                foreach (int n in blocks)
+                {
+                    if (n == this.RootBlock.Index)
+                        continue;
+                    BasicBlock<N, D> b = this.AllBlocks[n];
+                    if (b.PredecessorEdges == null)
```

Comment:
Isn't it a `Debug.Assert` instead ? We ensure that predecessor edges are set with previous call to `SetupPredecessorEdgesFromRoot`.

Reply:
done in 
ff21062